Description: reduce the rust debuginfo level on selected architectures where
 compiling with debuginfo=2 causes the OOM killer to interrupt the build on
 launchpad builders. Initially this was only on 32 bit architectures, but with
 firefox 63 it started happening frequently on arm64 and ppc64el too.

--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -1522,8 +1522,8 @@ def rustc_opt_level(opt_level_option, mo
         return '1' if moz_optimize.optimize else '0'
 
 
-@depends(rustc_opt_level, debug_rust, '--enable-debug-symbols')
-def rust_compile_flags(opt_level, debug_rust, debug_symbols):
+@depends(rustc_opt_level, debug_rust, '--enable-debug-symbols', host)
+def rust_compile_flags(opt_level, debug_rust, debug_symbols, host):
     # Cargo currently supports only two interesting profiles for building:
     # development and release. Those map (roughly) to --enable-debug and
     # --disable-debug in Gecko, respectively.
@@ -1546,6 +1546,8 @@ def rust_compiler_flags(opt_level, debug
 
     if debug_symbols:
         debug_info = '2'
+        if host.bitness == 32 or host.cpu == 'aarch64' or host.cpu == 'ppc64':
+            debug_info = '1'
 
     opts = []
 
